### ISSUE TITLE
XDG Base Directory support

### ISSUE THREAD

**Issue by purpleKarrot** [sqitchers/sqitch#468]
Instead of (or in addition to) `~/.sqitch/`, search the user configuration in `$XDG_CONFIG_HOME/sqitch/`, `~/.config/sqitch/`.

https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html

**Comment by theory** [comment]
It's not just the Sqitch config file, but also configuration files used by each engine.

**Comment by purpleKarrot** [comment]
Correct. It is about the whole directory. That's why I put a `/` at the end.

**Comment by theory** [comment]
Well, we could easily teach to favor `$XDG_CONFIG_HOME` over `$HOME` when looking for files, but I don't know whether and how well each engine supports that environment variable when reading its config files. We'd need to do an inventory, I expect, and probably send feature requests similar to this one to the engines that don't pay attention to `$XDG_CONFIG_HOME`. Snowflake comes to mind; it reads `$WORKSPACE` and `$HOME`.

**Comment by lhanson** [comment]
Please implement, respecting users' preferences is the right thing to do.

**Comment by theory** [comment]
Not if the database clients (`psql`, `snowsql`, `sqlite3`, etc. don’t support it.
